From f055f355faf1991ef4e6b3c3517f8f2fc247805e Mon Sep 17 00:00:00 2001
Message-Id: <f055f355faf1991ef4e6b3c3517f8f2fc247805e.1649665190.git.geliang.tang@suse.com>
From: Alexei Starovoitov <ast@kernel.org>
Date: Fri, 6 Nov 2020 12:33:46 -0800
Subject: [PATCH] selftests/bpf: Fix selftest build with old libc
Git-commit: f055f355faf1991ef4e6b3c3517f8f2fc247805e
Patch-mainline: v5.11-rc1
References: bsn#127
Modified-by-SEL: No


pidfd_open was added in 2019. Some versions of libc library don't define it.
Define it manually if it's not available.

Reported-by: Sergei Iudin <siudin@fb.com>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/testing/selftests/bpf/prog_tests/test_local_storage.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/testing/selftests/bpf/prog_tests/test_local_storage.c b/tools/testing/selftests/bpf/prog_tests/test_local_storage.c
index 5fda45982be0..fcca7ba1f368 100644
--- a/tools/testing/selftests/bpf/prog_tests/test_local_storage.c
+++ b/tools/testing/selftests/bpf/prog_tests/test_local_storage.c
@@ -12,6 +12,10 @@
 #include "local_storage.skel.h"
 #include "network_helpers.h"
 
+#ifndef __NR_pidfd_open
+#define __NR_pidfd_open 434
+#endif
+
 static inline int sys_pidfd_open(pid_t pid, unsigned int flags)
 {
 	return syscall(__NR_pidfd_open, pid, flags);
-- 
2.34.1

